src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,417 @@
+    this.value = this.defaultValue;
+  }
+
+  onLabelClick(): void {
+    this.setFocus();
+  }
+
+  handleInput = (event: InputEvent): void => {
+    this.value = event.target["value"];
+    this.calciteTextareaInput.emit();
+  };
+
+  handleChange = (): void => {
+    this.calciteTextareaChange.emit();
+  };
+
+  footerTrailingSlotChangeHandler = (event: Event): void => {
+    this.trailingSlotElements = slotChangeGetAssignedElements(event);
+    if (this.disabled) {
+      this.disablePointerEvents(this.disabled, this.trailingSlotElements);
+    }
+  };
+
+  footerLeadingSlotChangeHandler = (event: Event): void => {
+    this.leadingSlotElements = slotChangeGetAssignedElements(event);
+    if (this.disabled) {
+      this.disablePointerEvents(this.disabled, this.leadingSlotElements);
+    }
+  };
+
+  contentSlotChangeHandler = (): void => {
+    if (!this.value) {
+      const nodes = this.el.childNodes;
+      nodes.forEach((el) => {
+        if (el.nodeName === "#text") {
+          this.value = el.nodeValue.trim();
+        }
+      });
+    }
+  };
+
+  renderCharacterLimit = (): VNode => {
+    return this.maxlength ? (
+      <span class={CSS.characterLimit}>
+        <span class={{ [CSS.characterOverlimit]: this.value?.length > this.maxlength }}>
+          {this.getLocalizedCharacterLength()}
+        </span>
+        {"/"}
+        {numberStringFormatter.localize(this.maxlength.toString())}
+      </span>
+    ) : null;
+  };
+
+  getLocalizedCharacterLength(): string {
+    numberStringFormatter.numberFormatOptions = {
+      locale: this.effectiveLocale,
+      numberingSystem: this.numberingSystem
+    };
+    return numberStringFormatter.localize(this.value?.length.toString());
+  }
+
+  disablePointerEvents(disabled: boolean, slottedElements: Element[]): void {
+    if (!!slottedElements?.length) {
+      slottedElements.forEach((el: HTMLElement) => {
+        el.style.pointerEvents = disabled ? "none" : "auto";
+      });
+    }
+  }
+
+  resizeObserver = createObserver("resize", () => {
+    const { width: textareaWidth, height: textareaHeight } =
+      this.textareaEl.getBoundingClientRect();
+    const { width: elWidth, height: elHeight } = this.el.getBoundingClientRect();
+    const footerHeight = this.footerEl?.getBoundingClientRect().height;
+
+    if (this.footer && this.footerEl) {
+      this.footerEl.style.width = `${textareaWidth}px`;
+    }
+
+    if (
+      textareaWidth &&
+      (elWidth !== textareaWidth || elHeight !== textareaHeight + (footerHeight || 0))
+    ) {
+      this.setHeightAndWidthToAuto();
+    }
+  });
+
+  syncHiddenFormInput(input: HTMLInputElement): void {
+    input.setCustomValidity("");
+    if (this.value?.length > this.maxlength) {
+      input.setCustomValidity(this.messages.overLimit);
+    }
+    if (this.invalid) {
+      input.setCustomValidity(this.messages.invalid);
+    }
+  }
+
+  setHeightAndWidthToAuto(): void {
+    this.timeOutId = window.setTimeout(() => {
```

Comment:
Do you need to use `window.setTimeout` or will just `setTimeout` work?

Maybe rename `timeOutId` to `resizeTimeout`?

Reply:
I think the timeout should be cleared right before being created.

Reply:
> Do you need to use `window.setTimeout` or will just `setTimeout` work?


both setTimeout and window.setTimeout works here, I see other components are using `window.setTimeout` and followed it.

Reply:
> Maybe rename `timeOutId` to `resizeTimeout`?

makes sense, you mean resizeTimeoutId ?

Reply:
> I think the timeout should be cleared right before being created.

the reason behind not adding clearTimeout before creating it to set the height and with to auto while user is dragging the resize handle. If we clear it and user play around with the drag handle not fixing on a position , then the height & width wont be set to `auto` because the resize invokations would be quicker than the 100ms or 150ms timeout causing overlap with neighbor elements. For this use case, it is safe to clear it on `disconnectedCallback` because once the timeout is cleared `setHeightAndWidthToAuto` method wont be invoked further.

Reply:
Have you tried using `throttle` instead of a timeout here? It sounds like `throttle` would work with how you are describing.

Reply:
tried `throttle` with `leading:false` to avoid first invokation to execute and it is working as expected.